### PR TITLE
Task/bws20 master sign in

### PR DESCRIPTION
**Pull request by sofia0602**

### CODE REVIEW

File: frontend/src/hooks/use-app-form/use-app-form.hook.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ const useAppForm = <T extends FormControlValues = FormControlValues>({
     resolver: validationSchema
       ? getFormValidationResolver(validationSchema)
       : undefined,
+    mode: 'onBlur',
```
Comment (what1s1ove) [inline frontend/src/hooks/use-app-form/use-app-form.hook.ts]: Actually , we want it on submit (the default behaviour )

File: frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import style from './sign-wrapper-form.module.scss';
+import { Link } from 'components/common/common';
+import { AppRoute } from 'common/enums/enums';
+import { SignInForm } from '../components';
+import { MasterSignInDto } from 'common/types/types';
+const SignWrapperForm: React.FC = () => {
+  const onSubmit: any = (data: MasterSignInDto) => {
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx]: avoid any

File: frontend/src/common/types/form/form-control-mode.type.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { Mode } from 'react-hook-form';
+
+type FormMode = Mode;
+
+export { type FormMode };
```
Comment (what1s1ove) [inline frontend/src/common/types/form/form-control-mode.type.ts]: we don't want it for now

File: frontend/src/components/app/app.tsx [hunk 4]
```diff
@@ -31,6 +32,7 @@ const App: FC = () => {
           <Route path={AppRoute.SIGN_UP} element={<Auth />} />
         </Routes>
       </div>
+      <SignWrapperForm></SignWrapperForm>
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]: not sure that we need it here

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
+  const { control, errors, handleSubmit } = useAppForm<MasterSignInDto>({
+    defaultValues: DEFAULT_LOGIN_PAYLOAD,
+    validationSchema: masterSignInValidationSchema,
+  });
+
+  return (
+    <>
+      <h1>Sign ⚛️</h1>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        {/* <p> */}
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]: remove comments

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+        <Input
+          type={InputType.TEXT}
+          label="Name"
+          placeholder="Enter your name"
+          name={getNameOf<MasterSignInDto>('name')}
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]:
```suggestion
          name={getNameOf<MasterSignInDto>('email')}
```

File: shared/src/common/enums/validation/master/master.ts [hunk 7]
```diff
@@ -1 +1,2 @@
 export { MasterSignUpValidationMessage } from './master-sign-up.validation-message';
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/master/master.ts]:
```suggestion
export { MasterValidationMessage } from './master-sign-up.validation-message';
```
rename filename & enum to the `MasterValidationMessage` and use it for both sign-up and sign-in

File: shared/src/common/types/master/master-sign-in-dto.type.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+type MasterSignInDto = {
+  name: string;
```
Comment (what1s1ove) [inline shared/src/common/types/master/master-sign-in-dto.type.ts]:
```suggestion
  email: string;
```

File: shared/src/validation/master/master-sign-in/master-sign-in.validation-schema.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import * as Joi from 'joi';
+import { getNameOf } from '~/helpers/helpers';
+import { MasterSignInDto } from '~/common/types/types';
+import { MasterSignInValidationMessage } from '~/common/enums/enums';
+
+const masterSignIn = Joi.object({
+  [getNameOf<MasterSignInDto>('name')]: Joi.string()
```
Comment (what1s1ove) [inline shared/src/validation/master/master-sign-in/master-sign-in.validation-schema.ts]: should be an email

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import style from './sign-wrapper-form.module.scss';
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx]: actually, we don't need this component
with wrapper should be as `frontend/src/components/auth/auth.tsx`component

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 11]
```diff
@@ -1,16 +1,55 @@
...
+            name={getNameOf<MasterSignInDto>('password')}
+            control={control}
+            errors={errors}
+          />
+        </p>
+        <Button type={ButtonType.SUBMIT} label="Sign in" />
+        <div className={style.subtitle}>
+          By using SWS you are agreeing to our{' '}
+          <span className={style.link}>privacy policy</span>
+        </div>
```
Comment (Yaroslaww-1) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx:49]: I know that design has it, but you can drop privacy policy

File: frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import { SignInForm } from '../components';
+import { MasterSignInDto } from 'common/types/types';
+import { auth as authActions } from 'store/actions';
+import { useAppDispatch } from 'hooks/hooks';
+import Logo from '../../../../assets/img/logo.svg';
+const SignWrapperForm: React.FC = () => {
```
Comment (Yaroslaww-1) [inline frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx]: Please leave end lines between code components

File: frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <>
+      <div className={style.wrapper}>
+        <img src={Logo} alt="logo" className={style.logo} />
+        <SignInForm onSubmit={handleSignInSubmit} />
+      </div>
+    </>
+  );
+};
+export { SignWrapperForm };
```
Comment (Yaroslaww-1) [inline frontend/src/components/auth/components/sign-wrapper-form/sign-wrapper-form.tsx]:
```suggestion
import { SignInForm } from '../components';
import { MasterSignInDto } from 'common/types/types';
import { auth as authActions } from 'store/actions';
import { useAppDispatch } from 'hooks/hooks';
import Logo from '../../../../assets/img/logo.svg';

const SignWrapperForm: React.FC = () => {
  const dispatch = useAppDispatch();
  
  const handleSignInSubmit = (payload: MasterSignInDto): void => {
    dispatch(authActions.signIn(payload));
  };
  
  return (
    <>
      <div className={style.wrapper}>
        <img src={Logo} alt="logo" className={style.logo} />
        <SignInForm onSubmit={handleSignInSubmit} />
      </div>
    </>
  );
};

export { SignWrapperForm };
```

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts [hunk 14]
```diff
@@ -1,37 +1,34 @@
 import * as Joi from 'joi';
 import { getNameOf } from '~/helpers/helpers';
 import { MasterSignUpDto } from '~/common/types/types';
-import {
-  MasterSignUpValidationMessage,
-  MasterSignUpValidationRule,
-} from '~/common/enums/enums';
-
+import { MasterValidationMessage } from '~/common/enums/enums';
+import { MasterSignUpValidationRule } from '~/common/enums/validation/master/master-sign-up-validation-rule.enum';
```
Comment (what1s1ove) [inline shared/src/validation/master/master-sign-up/master-sign-up.validation-schema.ts]:
```suggestion
import { MasterValidationMessage, MasterSignUpValidationRule } from '~/common/enums/enums';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**